cmd/loadtest/loadtest.go

```diff
@@ -510,6 +521,62 @@ func printResults(lts []loadTestSample) {
 	log.Info().Uint64("numErrors", numErrors).Msg("Num errors")
 }
 
+func cleanHex(hexStr string) string {
+	// remove 0x suffix if found in the input string
+	return strings.Replace(hexStr, "0x", "", -1)
+}
+
+func getTxPoolSize(rpc *ethrpc.Client) (uint64, error) {
+	var status map[string]interface{}
+	err := rpc.Call(&status, "txpool_status")
+	if err != nil {
+		return 0, err
+	}
+	pendingHex, ok := status["pending"].(string)
+	if !ok {
+		return 0, fmt.Errorf("unable to read pending txpool size")
+	}
+	queuedHex, ok := status["queued"].(string)
+	if !ok {
+		return 0, fmt.Errorf("unable to read queued txpool size")
+	}
+
+	pendingTxPoolSize, err := strconv.ParseUint(cleanHex(pendingHex), 16, 64)
+	if err != nil {
+		return 0, fmt.Errorf("unable to parse pending txpool size: %v", err)
+	}
+	queuedTxPoolSize, err := strconv.ParseUint(cleanHex(queuedHex), 16, 64)
+	if err != nil {
+		return 0, fmt.Errorf("unable to parse queued txpool size: %v", err)
+	}
+
+	return (pendingTxPoolSize + queuedTxPoolSize), nil
+}
+
+func updateRateLimit(rl *rate.Limiter, rpc *ethrpc.Client, steadyStateQueueSize uint64, rateLimitIncrement uint64, cycleDuration time.Duration) {
+	ticker := time.NewTicker(cycleDuration)
+	defer ticker.Stop()
+
+	for range ticker.C {
+		txPoolSize, err := getTxPoolSize(rpc)
+		if err != nil {
+			log.Error().Err(err).Msg("Error getting txpool size")
+			return
+		}
+
+		if txPoolSize < steadyStateQueueSize {
+			// additively increment requests per second if txpool less than queue steady state
+			newRateLimit := rate.Limit(float64(rl.Limit()) + float64(rateLimitIncrement))
+			rl.SetLimit(newRateLimit)
+			log.Trace().Interface("New Rate Limit (RPS)", rl.Limit()).Interface("Current Tx Pool Size", txPoolSize).Interface("Steady State Tx Pool Size", steadyStateQueueSize).Msg("Increased rate limit")
```

Comment:
Ideally use the correct types rather than `Interface` . I don't know if these are the correct types but it would look something like this
```
log.Trace().Float64("New Rate Limit (RPS)", rl.Limit()).Uint64("Current Tx Pool Size", txPoolSize).Uint64("Steady State Tx Pool Size", steadyStateQueueSize).Msg("Increased rate limit")
```

Reply:
makes sense, will adjust